language/e2e_tests/src/account.rs

```diff
@@ -121,21 +123,51 @@ impl Account {
     // Helpers for transaction creation with Account instance as sender
     //
 
-    /// Returns a [`SignedTransaction`] with no arguments and this account as the sender.
+    /// Returns a [`SignedTransaction`] with a payload and this account as the sender.
+    ///
+    /// This is the most generic way to create a transaction for testing.
+    /// Max gas amount and gas unit price are ignored for WriteSet transactions.
     pub fn create_signed_txn(
         &self,
-        program: Vec<u8>,
+        payload: TransactionPayload,
         sequence_number: u64,
         max_gas_amount: u64,
         gas_unit_price: u64,
     ) -> SignedTransaction {
-        self.create_signed_txn_with_args(
-            program,
-            vec![],
-            sequence_number,
-            max_gas_amount,
-            gas_unit_price,
-        )
+        let raw_txn = match payload {
+            TransactionPayload::Program(program) => RawTransaction::new(
+                *self.address(),
+                sequence_number,
+                program,
+                max_gas_amount,
+                gas_unit_price,
+                Duration::from_secs(u64::max_value()),
+            ),
+            TransactionPayload::WriteSet(writeset) => {
+                RawTransaction::new_write_set(*self.address(), sequence_number, writeset)
+            }
+            TransactionPayload::Module(module) => RawTransaction::new_module(
```

Comment:
IMHO having only `RawTransaction::new` seems clearer to me rather than having `new_module`, `new_script` and `new`.

Reply:
well new takes a Program which we will remove at some point.
I thought about having new() only and passing a TransactionPayload - that could be an option - but wanted to go with the current model of having different "constructors".
I am not sure what to do and how much it matters.
Let me know if you are expecting a change or not

Reply:
This shouldn't be a big issue. I'm fine keeping the code as it is.